atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -425,7 +429,14 @@ private CassandraKeyValueServiceImpl(
         this.cassandraTables = new CassandraTables(clientPool, config);
         this.taskRunner = new TaskRunner(executor);
         this.cellLoader = CellLoader.create(clientPool, wrappingQueryRunner, taskRunner, runtimeConfigSupplier);
-        this.asyncCellLoader = AsyncCellLoader.create(cqlClient, executor, config.getKeyspaceOrThrow());
+        this.asyncCellLoader = AsyncCellLoader.create(
+                cqlClient,
+                createInstrumentedFixedThreadPool(
```

Comment:
I feel like you should inject this threadpool, when an internal product instantiates a lot of these it will probably want to reuse the threadpool

Reply:
Discussed for now and this solution is fine for now.

Reply:
It's fine *IF* we make sure we don't create the threadpool the same way the other one is created: with corePoolSize == maxPoolSize. Let's create with corePoolSize == 0

Reply:
Resolved.